### DISCUSSION TITLE
Using the tools as a 6809 cross-compiler?

### DISCUSSION THREAD

**Discussion by DoctorWkt** [Deek/CoCoC#19] [Q&A] [answered]
Hi, I see that most of the subdirectories have GNU Makefiles, and I've been able to build most of the tools on a Linux box. Is there a document that tells me where to install the resulting binaries so I can use this as a cross-compiler?

Should I use CC, CC2 or DCC as the front-end for the compiler passes? Can AR be built on a Linux box? What assembler does CoCoC use?

[Why? I'm building a 6809 SBC and I'm trying to find a C compiler and tools that, initially, can cross-compile but, eventually, run self-hosted on the SBC.]

Thanks, Warren

**Comment by Deek** [comment, marked as answer]
> Hi, I see that most of the subdirectories have GNU Makefiles, and I've been able to build most of the tools on a Linux box. Is there a document that tells me where to install the resulting binaries so I can use this as a cross-compiler?

Sorry for the late response, I don't look at the discussions all the time.

Not as such, mostly because as I was putting the system together I got really used to building more stuff as I ran into roadblocks and the whole thing kinda snowballed. 😁

In my workflow, I use a /dd/defs directory to hold the headers meant to be seen by dcpp, and a /dd/lib directory to hold the libraries -- you should be able to copy them from the Defs/ and Lib/ dirs in CoCoC.

There are a few files that make installs in ~/.dcc/ that parts of the compiler use, and I think all the compiler parts install themselves into ~/bin/.

> Should I use CC, CC2 or DCC as the front-end for the compiler passes? Can AR be built on a Linux box? What assembler does CoCoC use?

`dcc` is the right frontend to use once you've got the compiler stuff built. The compiler is currently set up to use `rma` and `rlink` to assemble and link binaries, versions of which have been ported to Linux.

I am currently working on adopting a multiplatform `rlink` into CoCoC, but that's not done yet. After that, I hope to tackle RMA.

> [Why? I'm building a 6809 SBC and I'm trying to find a C compiler and tools that, initially, can cross-compile but, eventually, run self-hosted on the SBC.]

That's pretty cool, let me know if there's something I can do to help with that.

**Comment by DoctorWkt** [comment]
Thanks for the reply. In the interim, I've found the [vbcc](http://www.compilers.de/vbcc.html) compiler and associated assembler and linker to make a good toolchain for the 6809. I've just uploaded the current state of my 6809 SBC to Github; have a look if you have time: [Nine-E](https://github.com/DoctorWkt/Nine-E).
